x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1137,6 +1148,235 @@ protected Expression rule(Expression e) {
+                        if (operation.left() instanceof Neg) {
+                            return operation.replaceChildren(List.of(operation.left().children().get(0), operation.right()));
+                        }
+                        if (operation.right() instanceof Neg) {
+                            return operation.replaceChildren(List.of(operation.left(), operation.right().children().get(0)));
+                        }
+                    }
+                }
+            } else if (e instanceof ArithmeticOperation) { // pulls up MUL/DIV - NEG - x ==> NEG - MUL/DIV - x
+                ArithmeticOperation operation = (ArithmeticOperation) e;
+                String opSymbol = operation.symbol();
+                if (MUL.symbol().equals(opSymbol) || DIV.symbol().equals(opSymbol)) {
+                    if (operation.left() instanceof Neg) {
+                        Neg neg = (Neg) operation.left();
+                        return new Neg(operation.source(), operation.replaceChildren(List.of(neg.children().get(0), operation.right())));
+                    }
+                    if (operation.right() instanceof Neg) {
+                        Neg neg = (Neg) operation.right();
+                        return new Neg(operation.source(), operation.replaceChildren(List.of(operation.left(), neg.children().get(0))));
+                    }
+                }
+            }
+            return e;
+        }
+    }
+
+    // Simplifies arithmetic expressions with fixed point fields in BinaryComparisons.
+    public static final class SimplifyComparisonsArithmetics extends OptimizerExpressionRule {
+        BiFunction<DataType, DataType, Boolean> typesCompatible;
+
+        public SimplifyComparisonsArithmetics(BiFunction<DataType, DataType, Boolean> typesCompatible) {
+            super(TransformDirection.UP);
+            this.typesCompatible = typesCompatible;
+        }
+
+        @Override
+        protected Expression rule(Expression e) {
+            return (e instanceof BinaryComparison) ? simplify((BinaryComparison) e) : e;
+        }
+
+        private Expression simplify(BinaryComparison bc) {
+            // optimize only once the expression has a literal on the right side of the binary comparison
+            if (bc.right() instanceof Literal) {
+                if (bc.left() instanceof ArithmeticOperation) {
+                    return SimplifyOperation.simplify(bc, typesCompatible);
+                } else if (bc.left() instanceof Neg) {
+                    return reduceNegation(bc);
+                }
+            }
+            return bc;
+        }
+
+        private static Expression reduceNegation(BinaryComparison bc) {
+            Literal bcLiteral = (Literal) bc.right();
+            Expression neg = safeMaybeFold(new Neg(bcLiteral.source(), bcLiteral));
+            return neg == null ? bc : bc.reverse().replaceChildren(List.of(bc.left().children().get(0), neg));
+        }
```

Comment:
Some renaming and inlining would make this easier to read and reason about. The `safeMaybeFold` is just a really simple `try-catch` that you kind of mimic with the `null` check anyways.

```suggestion
        private static Expression reduceNegation(BinaryComparison bc) {
            Literal oldRight = (Literal) bc.right();
            Neg newRight = new Neg(oldRight.source(), oldRight);
            Expression newLeft = ((Neg) bc.left()).field();
            try {
                newRight = new Literal(newRight.source(), newRight.fold(), newRight.dataType());
            } catch (Exception e) {
                // cannot optimize the negation, either ArithmeticException or DateTimeException
                // but safe to just turn off the optmisation should any exception arise
                return bc;
            }
            return bc.reverse().replaceChildren(List.of(newLeft, newRight));
        }
```

Reply:
`safeMaybeFold` performs an additional check too and is called from other places too, so I'd personally prefer the existing more compact and code reusing way.